src/components/movie-card/index.jsx

```diff
@@ -1,79 +1,42 @@
 import React from 'react';
 import PropTypes from 'prop-types';
-import { DeleteModal } from '../delete-modal';
-import { ModalWindow } from '../modal-window';
-
+import { useModal } from '../context/modal-context';
 import styles from './MovieCard.module.css';
 
-export class MovieCard extends React.Component {
-    constructor(props) {
-        super(props);
-        this.state = {
-            showDeleteModal: false,
-            deleteModalHeight: 0,
-            showEditModal: false,
-            editModalHeight: 0
-        };
-        this.handleDeleteModalClose = this.handleDeleteModalClose.bind(this);
-        this.handleEditModalClose = this.handleEditModalClose.bind(this);
-        this.handleModalOpen = this.handleModalOpen.bind(this);
-    }
-
-    handleDeleteModalClose = (e) => {
-        const currentClass = e.target.className;
-        if (currentClass === 'modal-background' || currentClass === 'close') {
-            this.setState({showDeleteModal: false});
-        };
-    };
-
-    handleEditModalClose = (e) => {
-        const currentClass = e.target.className;
-        if (currentClass === 'modal-background' || currentClass === 'close') {
-            this.setState({showEditModal: false});
-        };
-    };
+export const MovieCard = (props) => {
+    const [modalObject, setModalObject] = useModal();
 
-    handleModalOpen = (e) => {
+    const handleModalOpen = (e) => {
         e.preventDefault();
         const buttonId = e.target.id;
         if(buttonId === 'delete-button'){
-            this.setState({
-                showDeleteModal: true,
-                deleteModalHeight: document.body.scrollHeight
-            });
+            setModalObject({'modalType': 'delete-modal'})
```

Comment:
I guess you need to transfer card `id` too. Otherwise, you can't delete a particular card.